wormhole-connect/src/hooks/useRoutesQuotesBulk.ts

```diff
@@ -0,0 +1,144 @@
+import { useState, useEffect, useMemo } from 'react';
+import { Chain, amount as sdkAmount, routes } from '@wormhole-foundation/sdk';
+
+import { getTokenDecimals } from 'utils';
+import config from 'config';
+import { toDecimals } from 'utils/balance';
+
+type RoutesQuotesBulkParams = {
+  sourceChain?: Chain;
+  sourceToken: string;
+  destChain?: Chain;
+  destToken: string;
+  amount: string;
+  nativeGas: number;
+};
+
+export type ParsedQuote = {
+  error?: string;
+  eta?: number;
+  receiveAmount: string;
+  receiveNativeAmount: number;
+  relayerFee?: number;
+};
+
+type HookReturn = {
+  quotesMap: Record<string, ParsedQuote | undefined>;
+  isFetching: boolean;
+};
+
+const defaultQuote: ParsedQuote = {
+  error: '',
+  eta: 0,
+  receiveAmount: '',
+  receiveNativeAmount: 0,
+  relayerFee: 0,
+};
+
+const parseQuote = (
+  quote: routes.QuoteResult<routes.Options>,
+  sourceChain: Chain,
+): ParsedQuote => {
+  if (!quote.success) {
+    return {
+      ...defaultQuote,
+      error: quote.error.message,
+    };
+  }
+
+  const receiveAmount = sdkAmount
+    .whole(quote.destinationToken.amount)
+    .toString();
+  const receiveNativeAmount = quote.destinationNativeGas
+    ? sdkAmount.whole(quote.destinationNativeGas)
+    : 0;
+  const eta = quote.eta;
+  let relayerFee: number | undefined = undefined;
+
+  if (quote.relayFee) {
+    const { token, amount } = quote.relayFee;
+    const feeToken = config.sdkConverter.toTokenIdV1(token);
+    const decimals = getTokenDecimals(sourceChain, feeToken);
+    relayerFee = Number.parseFloat(toDecimals(amount.amount, decimals, 6));
+  }
+
+  return {
+    receiveAmount,
+    receiveNativeAmount,
+    eta,
+    relayerFee,
+  };
+};
+
+const useRoutesQuotesBulk = (
+  routes: string[],
+  params: RoutesQuotesBulkParams,
+): HookReturn => {
+  const [isFetching, setIsFetching] = useState(false);
+  const [quotes, setQuotes] = useState<ParsedQuote[]>([]);
+
+  useEffect(() => {
+    let unmounted = false;
+    if (
+      !params.sourceChain ||
+      !params.sourceToken ||
+      !params.destChain ||
+      !params.destToken ||
+      !params.amount
+    ) {
+      return;
+    }
+
+    // Forcing TS to infer that fields are non-optional
+    const rParams = params as Required<RoutesQuotesBulkParams>;
+    const promises = config.routes.computeMultipleQuotes(routes, rParams);
+
+    setIsFetching(true);
+    Promise.allSettled(promises).then((results) => {
```

Comment:
IMO you could move more of this logic into the `RouteOperator` method. I also notice you didn't type the return value of `computeMultipleQuotes`. What do you think about this?

- move `ParsedQuote` type into `routes/operator.ts` and import it from there (since multiple components import it)
- move the parsing logic into there as well
- make it so the return type of `computeMultipleQuotes` is `Async<ParsedQuote[]>` so you can simply just `await` that and have your list of quotes in here.